src/main/sd/init.c

```diff
@@ -0,0 +1,310 @@
+#include "sd.h"
+#include "debug.h"
+#include "pin.h"
+#include "spi.h"
+
+static bool sd_card_initialized = false;
```

Comment:
Maybe we shouldn't use global variables. Instead we could pass to init function a struct which we could fill.

Reply:
In my opinion this global varialbe is necessary. When program starts, card is not initialized. After initialization variable changes to true and any other attepmpt to initialize card will be prevented.